stats/stats-server/src/read_service.rs

```diff
@@ -83,8 +83,14 @@ impl StatsService for ReadService {
             })
             .collect();
 
-        // remove last data point, because it can be partially updated
-        data.pop();
+        let settings =
+            self.charts.settings.get(&request.name).ok_or_else(|| {
+                tonic::Status::not_found(format!("chart {} not found", request.name))
+            })?;
```

Comment:
maybe move this code to line#66?  because it looks like double check the same thing